source/exe/terminate_handler.cc

```diff
@@ -2,14 +2,28 @@
 
 #include <cstdlib>
 
+#include "envoy/common/exception.h"
 #include "source/common/common/logger.h"
 #include "source/server/backtrace.h"
 
+#include "absl/strings/str_format.h"
+
 namespace Envoy {
 
 std::terminate_handler TerminateHandler::logOnTerminate() const {
   return std::set_terminate([]() {
-    ENVOY_LOG(critical, "std::terminate called! (possible uncaught exception, see trace)");
+    auto currentException = std::current_exception();
+    if (currentException) {
+      try {
+        std::rethrow_exception(currentException);
+      } catch (const EnvoyException& e) {
+        ENVOY_LOG(critical, absl::StrFormat("std::terminate called! Uncaught EnvoyException '%s', see trace.", e.what()));
```

Comment:
nit: as much as I prefer using absl for this: you can just use the built-in fmtlib support in ENVOY_LOG in this case:
```
ENVOY_LOG(critical, "std::terminate called! Uncaught EnvoyException {}, see trace.", e.what());
```

Reply:
done